api/v1alpha1/hostdevicenetwork_webhook.go

```diff
@@ -0,0 +1,92 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"regexp"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var hostdevicenetworklog = logf.Log.WithName("hostdevicenetwork-resource")
+
+func (r *HostDeviceNetwork) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+//nolint:lll
+//+kubebuilder:webhook:path=/validate-mellanox-com-v1alpha1-hostdevicenetwork,mutating=false,failurePolicy=fail,sideEffects=None,groups=mellanox.com,resources=hostdevicenetworks,verbs=create;update,versions=v1alpha1,name=vhostdevicenetwork.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &HostDeviceNetwork{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *HostDeviceNetwork) ValidateCreate() error {
+	hostdevicenetworklog.Info("validate create", "name", r.Name)
+
+	return r.validateHostDeviceNetwork()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *HostDeviceNetwork) ValidateUpdate(_ runtime.Object) error {
+	hostdevicenetworklog.Info("validate update", "name", r.Name)
+
+	return r.validateHostDeviceNetwork()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *HostDeviceNetwork) ValidateDelete() error {
+	hostdevicenetworklog.Info("validate delete", "name", r.Name)
+
+	// TODO(user): fill in your validation logic upon object deletion.
+	return nil
+}
+func (r *HostDeviceNetwork) validateHostDeviceNetwork() error {
+	var allErrs field.ErrorList
+	// Validate ResourceName
+	resourceName := r.Spec.ResourceName
+	if !isValidHostDeviceNetworkResourceName(resourceName) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("Spec"), resourceName,
+			"Invalid Resource name, it must consist of alphanumeric characters, '-', '_' or '.', "+
+				"and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', "+
+				"regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')"))
+	}
+
+	if len(allErrs) == 0 {
```

Comment:
this list can contain only 1 or 0 elements, so we don't need to check it's length to simplify the flow:
```
if !isValidHostDeviceNetworkResourceName(resourceName) {
	allErrs :=  make([]field.ErrorList, 1]
	allErrs = append(allErrs, field.Invalid(field.NewPath("Spec"), resourceName,
			"Invalid Resource name, it must consist of alphanumeric characters, '-', '_' or '.', "+
				"and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', "+
				"regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')"))
	return apierrors.NewInvalid(
		schema.GroupKind{Group: "mellanox.com", Kind: "HostDeviceNetwork"},
		r.Name, allErrs)
}
return nil		
```

Reply:
Done